pkgs/applications/audio/easyabc/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, python39, makeWrapper, fluidsynth, soundfont-fluid, wrapGAppsHook, abcmidi, abcm2ps, ghostscript }:
+
+# requires python39 due to https://stackoverflow.com/a/71902541
+python39.pkgs.buildPythonApplication rec {
+  pname = "easyabc";
+  version = "1.3.8.6";
+
+  src = fetchFromGitHub {
+    owner = "jwdj";
+    repo = pname;
+    rev = "6461b2c14280cb64224fc5299c31cfeef9b7d43c";
+    sha256 = "sha256-leC3A4HQMeJNeZXArb3YAYr2mddGPcws618NrRh2Q1Y=";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  propagatedBuildInputs = with python39.pkgs; [
+    cx_Freeze
+    wxPython_4_2
+    pygame
+  ];
+
+  # apparently setup.py only supports Windows and Darwin
+  # everything is very non-standard in this project
+  dontBuild = true;
+  format = "other";
+
+  # https://discourse.nixos.org/t/packaging-mcomix3-python-gtk-missing-gsettings-schemas-issue/10190/2
+  strictDeps = false;
+
+  installPhase = ''
+    mkdir -p $out/share/easyabc
+    mv * $out/share/easyabc
+
+    sed -i '46 i\lib_locations = ["${fluidsynth}/lib/libfluidsynth.so"]' $out/share/easyabc/fluidsynth.py
+    sed -i '3962 i\        default_soundfont_path = "${soundfont-fluid}/share/soundfonts/FluidR3_GM2-2.sf2"' $out/share/easyabc/easy_abc.py
+    sed -i '8371 i\            settings["gs_path"] = "${ghostscript}/bin/gs"' $out/share/easyabc/easy_abc.py
+
+    ln -s ${abcmidi}/bin/abc2midi $out/share/easyabc/bin/abc2midi
+    ln -s ${abcmidi}/bin/midi2abc $out/share/easyabc/bin/midi2abc
+    ln -s ${abcmidi}/bin/abc2abc $out/share/easyabc/bin/abc2abc
+    ln -s ${abcm2ps}/bin/abcm2ps $out/share/easyabc/bin/abcm2ps
+
+    makeWrapper ${python39.interpreter} $out/bin/easyabc \
+      --set PYTHONPATH "$PYTHONPATH:$out/share/easyabc" \
+      --add-flags "-O $out/share/easyabc/easy_abc.py"
+  '';
+
+  meta = with lib; {
+    description = "abc music notation editor";
+    homepage = "https://easyabc.sourceforge.net/";
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ mausch ];
+  };
```

Comment:
```suggestion
  meta = {
    description = "ABC music notation editor";
    homepage = "https://easyabc.sourceforge.net/";
    license = lib.licenses.gpl2Plus;
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ mausch ];
  };
```

Reply:
The criteria are

- avoid `with`
- never use `with` inside another `with`

This is on my todo list of "code cleanup", as I have said above :)

Reply:
Thing is, I've been asked to use `with` before when I wasn't using it: https://github.com/NixOS/nixpkgs/pull/165183#discussion_r832828406

It's very frustrating for contributors that reviewers have wildly different criteria.

Reply:
The problems with `with` are exposed below:

https://nix.dev/recipes/best-practices#with-scopes

Reply:
Applied in https://github.com/NixOS/nixpkgs/pull/237197/commits/5fa3e97bef6ce3b29331a1d0cfc006132d3c79ed